contrib/pg_tde/src/catalog/tde_principal_key.c

```diff
@@ -598,12 +615,52 @@ pg_tde_set_principal_key_internal(char *principal_key_name, int is_global, char
 		providerOid = GLOBAL_DATA_TDE_OID;
 		dbOid = GLOBAL_DATA_TDE_OID;
 	}
+
+	if (is_global == 3)			/* using a globla provider for the default
+								 * encryption setting */
+	{
+		providerOid = GLOBAL_DATA_TDE_OID;
+		dbOid = DEFAULT_DATA_TDE_OID;
+
+		/* Do we already have a default key? If yes, look up the name of it */
+		LWLockAcquire(tde_lwlock_enc_keys(), LW_SHARED);
+		existingDefaultKey = GetPrincipalKeyNoDefault(dbOid, LW_SHARED);
+		if (existingDefaultKey != NULL)
+		{
+			memcpy(&existingKeyCopy, existingDefaultKey, sizeof(TDEPrincipalKey));
+		}
+		LWLockRelease(tde_lwlock_enc_keys());
+	}
+
 	success = set_principal_key_with_keyring(principal_key_name,
 											 provider_name,
 											 providerOid,
 											 dbOid,
 											 ensure_new_key);
 
+	if (is_global == 3 && existingDefaultKey != NULL)
+	{
+		/* In the previos step, we marked a new default provider */
+		/* Now we have to start using the new principal key in all databases */
+		/*
+		 * This violates the "only make changes to the actual database"
+		 * principle,
+		 */
+		/*
+		 * But there's no really good way to do this unless we want to move
+		 * this to a command line tool.
+		 */
```

Comment:
Why not merge this comment into one comment block?

Reply:
This was a pgindent change, fixed.